pkg/testutil/response_handlers.go

```diff
@@ -0,0 +1,39 @@
+package testutil
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+func init() {
+	knownHandlers = make(map[string]ResponseHandler)
+
+	RegisterResponseHandler("application/json", jsonHandler)
+}
+
+var (
+	knownHandlersMu sync.Mutex
+	knownHandlers   map[string]ResponseHandler
+)
+
+type ResponseHandler func(raw []byte, obj interface{}) error
```

Comment:
It would be useful to also pass the content type into the handler, as it may have useful information.

Reply:
thanks, done.